cli/client/main.go

```diff
@@ -416,14 +413,33 @@ func mainWithStatusCode() int {
 	}
 
 	hostnameIsAnIP := net.ParseIP(hostname) != nil
-
-	port := configPort
-	if port == -1 && urlPort != "" {
-		port, err = strconv.Atoi(urlPort)
-		if err != nil {
-			log.Error().Msgf("invalid port number: %s: %s", urlPort, err)
+	// *** Port selection ***
+	// There are six scenarios:
+	//
+	// | CLI port | parse | config port | program behaviour    |
+	// | -------- | ----- | ----------- | -------------------- |
+	// |    yes   |  nok  |     yes     | throw error and exit |
+	// |    yes   |  nok  |     no      | throw error and exit |
+	// |    yes   |  ok   |     yes     | use CLI port         |
+	// |    yes   |  ok   |     no      | use CLI port         |
+	// |    no    |  ---  |     yes     | use config port      |
+	// |    no    |  ---  |     no      | use default port     |
+
+	var port int
+	if urlPort != "" {
+		if parsedPort, err := strconv.Atoi(urlPort); err != nil || parsedPort > 0xffff {
+			// use WithLevel(zerolog.FatalLevel) to log a fatal level, but let us handle
```

Comment:
Would you mind inverting the two parts if this branch ? Putting the non-error case in the `if` part and the error case in the `else` part ? I think it makes it more readable because the default flow goes towards what we want to achieve. But maybe it is just me ? If you have good reasons to let it like that, let me know.

Reply:
Sure, will do. I think this is more a go habit, as most of my `if` statements are used to check for the absence of errors :smile: